### PR TITLE
fix(aws-certificatemanager): no unused props for DnsValidatedCertificate

### LINKED ISSUE #22739: aws-certificatemanager: DnsValidatedCertificate docs say that the default validation method is email
### Describe the issue

Documentation for the props for`DnsValidatedCertificate` say:

> validation?
>
> Type: [CertificateValidation](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_certificatemanager.CertificateValidation.html) (optional, default: CertificateValidation.fromEmail())
>
> How to validate this certificate.

which is obviously wrong, because its DNS validated.

I think this is just because `interface DnsValidatedCertificateProps extends CertificateProps`, and presumably this is a reasonable option in `CertificateProps`.

Solutions I can think of:
- use some `Omit<>` to remove `validation` from `DnsValidatedCertificateProps`
- set `validation: never` on `DnsValidatedCertificateProps`
- factor out some `BaseCertificateProps` from `CertificateProps` that doesn't include `validation` and have `DnsValidatedCertificateProps` extend that instead.

### Links

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_certificatemanager.DnsValidatedCertificate.html#validation

### PR DESCRIPTION
**Pull request by plumdog**
Fixes #22739.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

n/a

### New Features

n/a

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-certificatemanager/lib/certificate-base.ts [hunk 1]
```diff
@@ -3,6 +3,53 @@ import { Statistic } from '@aws-cdk/aws-cloudwatch';
 import { Duration, Resource } from '@aws-cdk/core';
 import { ICertificate } from './certificate';
 
+/**
+ * Shared properties for certificates
+ *
+ * @internal
+ */
+export interface BaseCertificateProps {
```
Comment (plumdog) [inline packages/@aws-cdk/aws-certificatemanager/lib/certificate-base.ts:11]: Q: should this be `CertificateBaseProps`?

File: packages/@aws-cdk/aws-certificatemanager/test/integ.dns-validated-certificate.ts [hunk 2]
```diff
@@ -36,7 +36,6 @@ const hostedZone = PublicHostedZone.fromHostedZoneAttributes(stack, 'HostedZone'
 const cert = new DnsValidatedCertificate(stack, 'Certificate', {
   domainName,
   hostedZone,
-  validation: CertificateValidation.fromDns(hostedZone),
```
Comment (plumdog) [inline packages/@aws-cdk/aws-certificatemanager/test/integ.dns-validated-certificate.ts:39]: This certainly feels like it should be redundant (as the `fromDns` is covered by virtue of being a `DnsValidateCertificate`, and the `hostedZone` is passed as its own prop). And couldn't see it getting used in `dns-validated-certificate.ts` anywhere.

**Comment by plumdog** [comment]
I don't think this warrants a change to an integration test. The fix is to remove some options that get ignored, so I don't think there's anything new to test, but rather just want for any existing tests to still pass.

**Review by comcalvi — COMMENTED** [review]
This is a good change but unfortunately it is a breaking change. Anybody using `DnsValidatedCertificate` with `validation` is going to be broken by this. 

#21982 will deprecate `DnsValidatedCertificate`, so we should ensure that this change gets added when that is merged.

**Pull request closed without merging**